salt/dex/dex.yaml

```diff
@@ -0,0 +1,136 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: dex
+  namespace: kube-system
+  labels:
+    kubernetes.io/cluster-service: "true"
+---
+# Map the LDAP Administrators role to the Kubernetes system:masters group
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: system:dex
+subjects:
+- kind: ServiceAccount
+  name: dex
+  namespace: kube-system
+roleRef:
+  kind: ClusterRole
+  name: cluster-admin
+  apiGroup: rbac.authorization.k8s.io
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: dex
+  namespace: kube-system
+data:
+  config.yaml: |
+    issuer: "https://{{ pillar['api']['server']['external_fqdn'] }}:{{ pillar['dex']['node_port'] }}"
+    storage:
+      type: kubernetes
+      config:
+        inCluster: true
+    web:
+      https: 0.0.0.0:5556
+      tlsCert: /etc/dex/tls/dex.crt
+      tlsKey: /etc/dex/tls/dex.key
+    connectors:
+    - type: ldap
+      id: ldap
+      name: LDAP
+      config:
+        host: "{{ pillar['dashboard'] }}:389"
+        startTLS: true
+        bindDN: cn=admin,{{ pillar['ldap_internal_infra_domain'] }}
+        bindPW: admin
+        rootCA: {{ pillar['ssl']['ca_file'] }}
+        userSearch:
+          baseDN: ou=People,{{ pillar['ldap_internal_infra_domain'] }}
+          filter: "(objectClass=inetOrgPerson)"
+          username: mail
+          idAttr: DN
+          emailAttr: mail
+          nameAttr: cn          
+        groupSearch:
+          baseDN: ou=Groups,{{ pillar['ldap_internal_infra_domain'] }}
+          filter: "(objectClass=groupOfUniqueNames)"
+
+          userAttr: DN
+          groupAttr: uniqueMember
+
+          nameAttr: cn       
+
+    oauth2:
+      skipApprovalScreen: true
+
+    staticClients:
+    - id: caasp-cli
+      redirectURIs:
+      - 'http://127.0.0.1'
+      name: "CaaSP CLI"
+      secret: "swac7qakes7AvucH8bRucucH"
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    app: dex
+  name: dex
+  namespace: kube-system
+spec:
+  replicas: 3
```

Comment:
It would be good if we could make this configurable, [as we do with the DNS replicas](https://github.com/kubic-project/salt/blob/master/salt/addons/addons/kubedns.yaml.jinja#L10).

Reply:
I'm not terribly comfortable with it being less than 3 replicas because of the outage it would cause for users if this were set to 1 and were to go down. Configurable is fine, as long as the minimum is 3.

Reply:
I agree that three is a good value. But I was thinking more on the case were three replicas are too many (ie, our devenv). I would set `replicas: {{ some pillar }}` in this file, and set the default pillar value to 3. Then the pillar could be modified manually (I use [my own script](https://github.com/inercia/caasp-tf/blob/master/resources/common/caaspctl) for this kind of things, with `caaspctl pillar set ...`).